### PR TITLE
authentication

### PR DESCRIPTION
**Pull request by pahans**

### CODE REVIEW

File: lib/apm.js [hunk 1]
```diff
@@ -67,20 +69,73 @@ Apm.connect = function(appId, appSecret, options) {
...
+      var retry = new Retry({
+        minCount: 0, // don't do any immediate retries
+        baseTimeout: 5*1000 // start with 30s
+      });
+      if(++authCheckFailures < 5 ){
+        retry.retryLater(authCheckFailures, function(){
+          pingToCheckAuth(callback)
+        });
+      } else {
+
```
Comment (arunoda) [inline lib/apm.js]: Shouldn't we add some console.log here.

File: lib/apm.js [hunk 2]
```diff
@@ -67,20 +69,73 @@ Apm.connect = function(appId, appSecret, options) {
...
+        retryPingToCheckAuth();
+        //rety
+      }
+
+    });
+
+
+    function retryPingToCheckAuth(){
+      console.log('retrying to authenticate apm');
+      var retry = new Retry({
```
Comment (arunoda) [inline lib/apm.js]: Move this Retry logic out of this function.
Set auth retry forever.

File: lib/apm.js [hunk 3]
```diff
@@ -67,20 +69,73 @@ Apm.connect = function(appId, appSecret, options) {
     }
 
     function tryAgain(err) {
-      if(++retries < 5) {
-        Meteor.setTimeout(callHTTP, 1000 * retries);
+      console.log('retrying to send payload to APM server')
+      var retry = new Retry({
```
Comment (arunoda) [inline lib/apm.js]: Move this retry logic into out of this function.

**Comment by pahans** [comment]
please check now.

**Pull request merged**